code/src/components/StartPage.js

```diff
@@ -0,0 +1,45 @@
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import "antd/dist/antd.css";
+
+import labyrinth, { generateGame } from '../reducers/labyrinth';
+
+import { CardContainer, Title, TitleBar, Content, SecondaryText, Button} from './Styles';
+
+const StartPage  = () => {
+    const [player, setPlayer] = useState('');
+    const [notValidated, setNotValidated] = useState(false);
+
+    const dispatch = useDispatch();
+
+    const onPlayerSet = () => {
+        if (player) {
+            dispatch(labyrinth.actions.setPlayer(player));
+            dispatch(generateGame());
+        } else {
+            setNotValidated(true)
+        }

```

Comment:
Nice way of validating the player has input a name - it could maybe be improved by adding a minimum number of characters for player?